contracts/contracts/BoosterLite.sol

```diff
@@ -0,0 +1,572 @@
+    ) external nonReentrant {
+        require(msg.sender == feeManager, "!auth");
+
+        uint256 total = _lockFees.add(_stakerFees).add(_callerFees).add(_platform);
+        require(total <= MaxFees, ">MaxFees");
+
+        require(_lockFees >= 300 && _lockFees <= 3500, "!lockFees");
+        require(_stakerFees >= 300 && _stakerFees <= 1500, "!stakerFees");
+        require(_callerFees >= 10 && _callerFees <= 100, "!callerFees");
+        require(_platform <= 200, "!platform");
+
+        lockIncentive = _lockFees;
+        stakerIncentive = _stakerFees;
+        earmarkIncentive = _callerFees;
+        platformFee = _platform;
+
+        emit FeesUpdated(_lockFees, _stakerFees, _callerFees, _platform);
+    }
+
+    /**
+     * @notice Set the address of the treasury (i.e. vlCVX)
+     */
+    function setTreasury(address _treasury) external {
+        require(msg.sender == feeManager, "!auth");
+        treasury = _treasury;
+
+        emit TreasuryUpdated(_treasury);
+    }
+
+    /// END SETTER SECTION ///
+
+    function poolLength() external view returns (uint256) {
+        return poolInfo.length;
+    }
+
+    /**
+     * @notice Called by the PoolManager (i.e. PoolManagerProxy) to add a new pool - creates all the required
+     *         contracts (DepositToken, RewardPool, Stash) and then adds to the list!
+     */
+    function addPool(
+        address _lptoken,
+        address _gauge,
+        uint256 _stashVersion
+    ) external returns (bool) {
+        require(msg.sender == poolManager && !isShutdown, "!add");
+        require(_gauge != address(0) && _lptoken != address(0), "!param");
+
+        //the next pool's pid
+        uint256 pid = poolInfo.length;
+
+        //create a tokenized deposit
+        address token = ITokenFactory(tokenFactory).CreateDepositToken(_lptoken);
+        //create a reward contract for crv rewards
+        address newRewardPool = IRewardFactory(rewardFactory).CreateCrvRewards(pid, token, _lptoken);
+        //create a stash to handle extra incentives
+        address stash = IStashFactory(stashFactory).CreateStash(pid, _gauge, staker, _stashVersion);
+
+        //add the new pool
+        poolInfo.push(
+            PoolInfo({
+                lptoken: _lptoken,
+                token: token,
+                gauge: _gauge,
+                crvRewards: newRewardPool,
+                stash: stash,
+                shutdown: false
+            })
+        );
+        gaugeMap[_gauge] = true;
+        //give stashes access to rewardfactory and voteproxy
+        //   voteproxy so it can grab the incentive tokens off the contract after claiming rewards
+        //   reward factory so that stashes can make new extra reward contracts if a new incentive is added to the gauge
+        if (stash != address(0)) {
+            poolInfo[pid].stash = stash;
+            IStaker(staker).setStashAccess(stash, true);
+            IRewardFactory(rewardFactory).setAccess(stash, true);
+        }
```

Comment:
the if statement seems to be redundant and so does setting the stash value of the struct. Stash is created earlier, so address shouldn't be 0

```suggestion
            IStaker(staker).setStashAccess(stash, true);
            IRewardFactory(rewardFactory).setAccess(stash, true);
```